runtime/Cargo.toml

```diff
@@ -27,6 +27,13 @@ opentelemetry = { workspace = true }
 tracing-opentelemetry = { workspace = true }
 rayon = { workspace = true }
 
+[features]
+default = []
+iouring = ["dep:io-uring"]
+
+[target.'cfg(target_os = "linux")'.dependencies]
```

Comment:
Should we align this with the feature rather than the target os?

Reply:
This is already conditionally included based on whether the `iouring` feature is enabled. If it is, we include the iouring dependency.

`[target.'cfg(target_os = "linux")'.dependencies]` means we only include the dependency if the target OS is linux. A reasonable person might think these facts imply that `cargo build --features iouring` should fail on a non-linux machine. But nope! Apparently it just silently doesn't include a necessary dependency? I'll have to include a pragma to explicitly make compilation fail if you use iouring on a non-linux platform.

See https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies